libraries/AP_Scripting/generator/description/bindings.desc

```diff
@@ -148,8 +148,7 @@ singleton RangeFinder method get_pos_offset_orient Vector3f Rotation'enum ROTATI
 
 include AP_Terrain/AP_Terrain.h
 
-depends AP_TERRAIN_AVAILABLE 1 Scripting requires terrain to be available
-
+singleton AP_Terrain depends defined(AP_TERRAIN_AVAILABLE) && AP_TERRAIN_AVAILABLE == 1
```

Comment:
The downside to this is that previously scripting implied that terrain was available, where as now your script has to cope with the idea that terrain may not be there. Do we need a form that is a hard requirement, as well as an optional feature, or do we want to embrace allowing terrain to just be removed from scripting builds?

Reply:
added a exist method so we can now cope with singletons not being there more gracefully.